storage/file.go

```diff
@@ -86,6 +89,18 @@ func (f FileServiceClient) DeleteShareIfExists(name string) (bool, error) {
 // deleteShare makes the call to Delete Share operation endpoint and returns
 // the response
 func (f FileServiceClient) deleteShare(name string) (*storageResponse, error) {
+	if err := f.checkForStorageEmulator(); err != nil {
```

Comment:
I'm a bit against adding this check here but I couldn't find a better place either. It appears like `GetFileService()` does not return an error.